tests/hikari/test_guilds.py

```diff
@@ -558,6 +864,92 @@ def test_make_banner_url_when_no_hash(self, model):
         model.banner_hash = None
         assert model.make_banner_url(ext="png", size=2048) is None
 
+    @pytest.mark.asyncio
+    async def test_fetch_roles(self, model):
+        model.app.rest.fetch_roles = mock.AsyncMock()
+
+        roles = await model.fetch_roles()
+
+        model.app.rest.fetch_roles.assert_awaited_once_with(123)
+        assert roles is model.app.rest.fetch_roles.return_value
+
+    @pytest.mark.asyncio
+    async def test_fetch_owner(self, model):
+        model.app.rest.fetch_member = mock.AsyncMock()
+
+        assert await model.fetch_owner() is model.app.rest.fetch_member.return_value
+        model.app.rest.fetch_member.assert_awaited_once_with(123, 1111)
+
+    @pytest.mark.asyncio
+    async def test_fetch_widget_channel(self, model):
+        mock_channel = mock.Mock(spec_set=channels_.GuildChannel)
+        model.app.rest.fetch_channel = mock.AsyncMock(return_value=mock_channel)
+
+        assert await model.fetch_widget_channel() is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(192729)
+
+    @pytest.mark.asyncio
+    async def test_fetch_widget_channel_when_None(self, model):
+        model.widget_channel_id = None
+
+        assert await model.fetch_widget_channel() is None
+
+    @pytest.mark.asyncio
+    async def test_fetch_rules_channel(self, model):
+        mock_channel = mock.Mock(spec_set=channels_.GuildTextChannel)
+        model.app.rest.fetch_channel = mock.AsyncMock(return_value=mock_channel)
+
+        assert await model.fetch_rules_channel() is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(123445)
+
+    @pytest.mark.asyncio
+    async def test_fetch_rules_channel_when_None(self, model):
+        model.rules_channel_id = None
+
+        assert await model.fetch_rules_channel() is None
+
+    @pytest.mark.asyncio
+    async def test_fetch_system_channel(self, model):
+        mock_channel = mock.Mock(spec_set=channels_.GuildTextChannel)
+        model.app.rest.fetch_channel = mock.AsyncMock(return_value=mock_channel)
+
+        assert await model.fetch_system_channel() is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(123888)
+
+    @pytest.mark.asyncio
+    async def test_fetch_system_channel_when_None(self, model):
+        model.system_channel_id = None
+
+        assert await model.fetch_system_channel() is None
+
+    @pytest.mark.asyncio
+    async def test_fetch_public_updates_channel(self, model):
+        mock_channel = mock.Mock(spec_set=channels_.GuildTextChannel)
```

Comment:
As I said, no need to spec_set these. A simple spec should be enough

```suggestion
        mock_channel = mock.Mock(channels_.GuildTextChannel)
```
(applies to all other mocks using spec_set too)